kentikapi/internal/api_payloads/saved_filter.go

```diff
@@ -0,0 +1,182 @@
+package api_payloads
+
+import (
+	"time"
+
+	"github.com/kentik/community_sdk_golang/kentikapi/internal/utils"
+	"github.com/kentik/community_sdk_golang/kentikapi/models"
+)
+
+type GetAllSavedFilterResponse []GetSavedFilterResponse
+
+func (p GetAllSavedFilterResponse) ToSavedFilters() ([]models.SavedFilter, error) {
+	var result []models.SavedFilter
+	err := utils.ConvertList(p, GetSavedFilterResponse.ToSavedFilter, &result)
+	if err != nil {
+		return nil, err
+	}
+	return result, err
+}
+
+type GetSavedFilterResponse struct {
+	SavedFilterPayload
+}
+
+type CreateSavedFilterRequest struct {
+	SavedFilterPayload
+}
+
+type CreateSavedFilterResponse struct {
+	SavedFilterPayload
+}
+
+type UpdateSavedFilterRequest struct {
+	SavedFilterPayload
+}
+
+type UpdateSavedFilterResponse struct {
+	SavedFilterPayload
+}
+
+type SavedFilterPayload struct {
+	ID                IntAsString    `json:"id,omitempty"`
+	CompanyID         models.ID      `json:"company_id,string,omitempty"`
+	FilterName        string         `json:"filter_name"`
+	FilterDescription string         `json:"filter_description"`
+	FilterLevel       string         `json:"filter_level,omitempty"`
+	CreatedDate       *time.Time     `json:"cdate,omitempty"`
```

Comment:
You can validate that payload received from API contains CreatedDate,UpdatedDate fields with struct tags response:"get,post,put" if they are expected to always be present. So dereferencing these fields with * doesnt end up with seg fault in case null is received